[2025-06-09T08:28:16.877+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: multi_region_processor_7.process_region_eu.process_discounts_eu manual__2025-06-09T08:25:44.177066+00:00 [queued]>
[2025-06-09T08:28:16.896+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: multi_region_processor_7.process_region_eu.process_discounts_eu manual__2025-06-09T08:25:44.177066+00:00 [queued]>
[2025-06-09T08:28:16.896+0000] {taskinstance.py:2170} INFO - Starting attempt 2 of 2
[2025-06-09T08:28:16.926+0000] {taskinstance.py:2191} INFO - Executing <Task(_PythonDecoratedOperator): process_region_eu.process_discounts_eu> on 2025-06-09 08:25:44.177066+00:00
[2025-06-09T08:28:16.933+0000] {standard_task_runner.py:60} INFO - Started process 107477 to run task
[2025-06-09T08:28:16.937+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'multi_region_processor_7', 'process_region_eu.process_discounts_eu', 'manual__2025-06-09T08:25:44.177066+00:00', '--job-id', '656', '--raw', '--subdir', 'DAGS_FOLDER/master2.py', '--cfg-path', '/tmp/tmpnea3vdz2']
[2025-06-09T08:28:16.939+0000] {standard_task_runner.py:88} INFO - Job 656: Subtask process_region_eu.process_discounts_eu
[2025-06-09T08:28:17.054+0000] {task_command.py:423} INFO - Running <TaskInstance: multi_region_processor_7.process_region_eu.process_discounts_eu manual__2025-06-09T08:25:44.177066+00:00 [running]> on host afcdbcf38bc0
[2025-06-09T08:28:17.221+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='data-team' AIRFLOW_CTX_DAG_ID='multi_region_processor_7' AIRFLOW_CTX_TASK_ID='process_region_eu.process_discounts_eu' AIRFLOW_CTX_EXECUTION_DATE='2025-06-09T08:25:44.177066+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-09T08:25:44.177066+00:00'
[2025-06-09T08:28:26.503+0000] {logging_mixin.py:188} INFO - Processing discounts for region eu
[2025-06-09T08:28:39.678+0000] {logging_mixin.py:188} INFO - +-----------+-------------+--------------------+-------------+--------------+--------------------+--------------------+-----------+-----------+-------------------+-------------------+---------+-------------------+-------+--------+
|discount_id|discount_code|       discount_name|discount_type|discount_value|minimum_order_amount|maximum_order_amount|usage_limit|usage_count|         start_date|           end_date|is_active|         created_at|_region| _source|
+-----------+-------------+--------------------+-------------+--------------+--------------------+--------------------+-----------+-----------+-------------------+-------------------+---------+-------------------+-------+--------+
|          1|    WELCOME10|New Customer Welcome|   percentage|          10.0|               50.00|              100.00|       1000|        234|2023-12-31 23:00:00|2024-12-31 22:59:59|        1|2025-06-08 19:22:57|     eu|postgres|
|          2|     SPRING20| Spring Sale 20% Off|   percentage|          20.0|              100.00|              200.00|        500|        156|2024-02-29 23:00:00|2024-03-31 22:59:59|        0|2025-06-08 19:22:57|     eu|postgres|
|          3|     FREESHIP|       Free Shipping|     shipping|           0.0|               75.00|               15.00|       2000|        567|2023-12-31 23:00:00|2024-12-31 22:59:59|        1|2025-06-08 19:22:57|     eu|postgres|
|          4|    SAVE50EUR|        Save 50 Euro|        fixed|          50.0|              200.00|               50.00|        300|         89|2024-01-31 23:00:00|2024-02-29 22:59:59|        0|2025-06-08 19:22:57|     eu|postgres|
|          5|BLACKFRIDAY30|Black Friday 30% Off|   percentage|          30.0|              150.00|              300.00|       1000|        678|2024-11-24 23:00:00|2024-11-29 22:59:59|        0|2025-06-08 19:22:57|     eu|postgres|
|          6|    STUDENT15|    Student Discount|   percentage|          15.0|               30.00|              150.00|          0|        123|2023-12-31 23:00:00|2024-12-31 22:59:59|        1|2025-06-08 19:22:57|     eu|postgres|
|          7|    LOYALTY25|Loyalty Member 25...|   percentage|          25.0|              100.00|              250.00|        500|        234|2023-12-31 23:00:00|2024-12-31 22:59:59|        1|2025-06-08 19:22:57|     eu|postgres|
|          8|     SUMMER15|      Summer Special|   percentage|          15.0|               80.00|              120.00|        750|        345|2024-05-31 23:00:00|2024-08-31 22:59:59|        1|2025-06-08 19:22:57|     eu|postgres|
|          9|     FIRST100| First 100 Customers|        fixed|          25.0|              100.00|               25.00|        100|        100|2023-12-31 23:00:00|2024-01-31 22:59:59|        0|2025-06-08 19:22:57|     eu|postgres|
|         10|    WEEKEND10|     Weekend Special|   percentage|          10.0|               60.00|               80.00|          0|        456|2023-12-31 23:00:00|2024-12-31 22:59:59|        1|2025-06-08 19:22:57|     eu|postgres|
+-----------+-------------+--------------------+-------------+--------------+--------------------+--------------------+-----------+-----------+-------------------+-------------------+---------+-------------------+-------+--------+
only showing top 10 rows
[2025-06-09T08:28:39.679+0000] {logging_mixin.py:188} INFO - Inserting into snowflake
[2025-06-09T08:28:49.999+0000] {logging_mixin.py:188} INFO - Error processing discounts for eu: An error occurred while calling o131.save.
: net.snowflake.client.jdbc.SnowflakeSQLException: Cannot perform CREATE TEMPSTAGE. This session does not have a current database. Call 'USE DATABASE', or use a qualified name.
	at net.snowflake.client.jdbc.SnowflakeUtil.checkErrorAndThrowExceptionSub(SnowflakeUtil.java:127)
	at net.snowflake.client.jdbc.SnowflakeUtil.checkErrorAndThrowException(SnowflakeUtil.java:67)
	at net.snowflake.client.core.StmtUtil.pollForOutput(StmtUtil.java:451)
	at net.snowflake.client.core.StmtUtil.execute(StmtUtil.java:354)
	at net.snowflake.client.core.SFStatement.executeHelper(SFStatement.java:487)
	at net.snowflake.client.core.SFStatement.executeQueryInternal(SFStatement.java:198)
	at net.snowflake.client.core.SFStatement.executeQuery(SFStatement.java:135)
	at net.snowflake.client.core.SFStatement.execute(SFStatement.java:767)
	at net.snowflake.client.core.SFStatement.execute(SFStatement.java:677)
	at net.snowflake.client.jdbc.SnowflakeStatementV1.executeQueryInternal(SnowflakeStatementV1.java:238)
	at net.snowflake.client.jdbc.SnowflakePreparedStatementV1.executeQuery(SnowflakePreparedStatementV1.java:117)
	at net.snowflake.spark.snowflake.JDBCWrapper.$anonfun$executePreparedQueryInterruptibly$1(SnowflakeJDBCWrapper.scala:205)
	at net.snowflake.spark.snowflake.JDBCWrapper.$anonfun$executeInterruptibly$2(SnowflakeJDBCWrapper.scala:252)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
[2025-06-09T08:28:50.320+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/decorators/base.py", line 241, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/master2.py", line 207, in process_table
    write_to_snowflake(df_transformed, table_name=table, region=region, sf_options=SF_OPTIONS)
  File "/opt/airflow/dags/master2.py", line 197, in write_to_snowflake
    .save()
  File "/home/airflow/.local/lib/python3.10/site-packages/pyspark/sql/readwriter.py", line 1461, in save
    self._jwrite.save()
  File "/home/airflow/.local/lib/python3.10/site-packages/py4j/java_gateway.py", line 1322, in __call__
    return_value = get_return_value(
  File "/home/airflow/.local/lib/python3.10/site-packages/pyspark/errors/exceptions/captured.py", line 179, in deco
    return f(*a, **kw)
  File "/home/airflow/.local/lib/python3.10/site-packages/py4j/protocol.py", line 326, in get_return_value
    raise Py4JJavaError(
py4j.protocol.Py4JJavaError: An error occurred while calling o131.save.
: net.snowflake.client.jdbc.SnowflakeSQLException: Cannot perform CREATE TEMPSTAGE. This session does not have a current database. Call 'USE DATABASE', or use a qualified name.
	at net.snowflake.client.jdbc.SnowflakeUtil.checkErrorAndThrowExceptionSub(SnowflakeUtil.java:127)
	at net.snowflake.client.jdbc.SnowflakeUtil.checkErrorAndThrowException(SnowflakeUtil.java:67)
	at net.snowflake.client.core.StmtUtil.pollForOutput(StmtUtil.java:451)
	at net.snowflake.client.core.StmtUtil.execute(StmtUtil.java:354)
	at net.snowflake.client.core.SFStatement.executeHelper(SFStatement.java:487)
	at net.snowflake.client.core.SFStatement.executeQueryInternal(SFStatement.java:198)
	at net.snowflake.client.core.SFStatement.executeQuery(SFStatement.java:135)
	at net.snowflake.client.core.SFStatement.execute(SFStatement.java:767)
	at net.snowflake.client.core.SFStatement.execute(SFStatement.java:677)
	at net.snowflake.client.jdbc.SnowflakeStatementV1.executeQueryInternal(SnowflakeStatementV1.java:238)
	at net.snowflake.client.jdbc.SnowflakePreparedStatementV1.executeQuery(SnowflakePreparedStatementV1.java:117)
	at net.snowflake.spark.snowflake.JDBCWrapper.$anonfun$executePreparedQueryInterruptibly$1(SnowflakeJDBCWrapper.scala:205)
	at net.snowflake.spark.snowflake.JDBCWrapper.$anonfun$executeInterruptibly$2(SnowflakeJDBCWrapper.scala:252)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

[2025-06-09T08:28:50.341+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=multi_region_processor_7, task_id=process_region_eu.process_discounts_eu, execution_date=20250609T082544, start_date=20250609T082816, end_date=20250609T082850
[2025-06-09T08:28:50.368+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 656 for task process_region_eu.process_discounts_eu (An error occurred while calling o131.save.
: net.snowflake.client.jdbc.SnowflakeSQLException: Cannot perform CREATE TEMPSTAGE. This session does not have a current database. Call 'USE DATABASE', or use a qualified name.
	at net.snowflake.client.jdbc.SnowflakeUtil.checkErrorAndThrowExceptionSub(SnowflakeUtil.java:127)
	at net.snowflake.client.jdbc.SnowflakeUtil.checkErrorAndThrowException(SnowflakeUtil.java:67)
	at net.snowflake.client.core.StmtUtil.pollForOutput(StmtUtil.java:451)
	at net.snowflake.client.core.StmtUtil.execute(StmtUtil.java:354)
	at net.snowflake.client.core.SFStatement.executeHelper(SFStatement.java:487)
	at net.snowflake.client.core.SFStatement.executeQueryInternal(SFStatement.java:198)
	at net.snowflake.client.core.SFStatement.executeQuery(SFStatement.java:135)
	at net.snowflake.client.core.SFStatement.execute(SFStatement.java:767)
	at net.snowflake.client.core.SFStatement.execute(SFStatement.java:677)
	at net.snowflake.client.jdbc.SnowflakeStatementV1.executeQueryInternal(SnowflakeStatementV1.java:238)
	at net.snowflake.client.jdbc.SnowflakePreparedStatementV1.executeQuery(SnowflakePreparedStatementV1.java:117)
	at net.snowflake.spark.snowflake.JDBCWrapper.$anonfun$executePreparedQueryInterruptibly$1(SnowflakeJDBCWrapper.scala:205)
	at net.snowflake.spark.snowflake.JDBCWrapper.$anonfun$executeInterruptibly$2(SnowflakeJDBCWrapper.scala:252)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
; 107477)
[2025-06-09T08:28:50.410+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-06-09T08:28:50.447+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
